src/libs/pool-data-provider/hooks/use-incentives-data-context.tsx

```diff
@@ -0,0 +1,292 @@
+              reserveIncentiveData: incentiveData,
+              totalLiquidity: formatBNInput(reserve.totalLiquidity, Number(reserve.decimals)),
+              liquidityIndex: formatBNInput(reserve.liquidityIndex, RAY_DECIMALS),
+              totalScaledVariableDebt: formatBNInput(
+                reserve.totalScaledVariableDebt,
+                Number(reserve.decimals)
+              ),
+              totalPrincipalStableDebt: formatBNInput(
+                reserve.totalPrincipalStableDebt,
+                Number(reserve.decimals)
+              ),
+              priceInMarketReferenceCurrency: formatBNInput(reserve.price.priceInEth, ETH_DECIMALS), // Will be replaced with marketReferencePrice/Decimals
+              decimals: Number(reserve.decimals),
+
+              aRewardTokenPriceInMarketReferenceCurrency: calculateRewardTokenPrice(
+                reserves,
+                incentiveData.aIncentiveData.rewardTokenAddress.toLowerCase(),
+                incentiveData.aIncentiveData.priceFeed,
+                incentiveData.aIncentiveData.priceFeedDecimals
+              ),
+
+              vRewardTokenPriceInMarketReferenceCurrency: calculateRewardTokenPrice(
+                reserves,
+                incentiveData.vIncentiveData.rewardTokenAddress.toLowerCase(),
+                incentiveData.vIncentiveData.priceFeed,
+                incentiveData.vIncentiveData.priceFeedDecimals
+              ),
+              sRewardTokenPriceInMarketReferenceCurrency: calculateRewardTokenPrice(
+                reserves,
+                incentiveData.sIncentiveData.rewardTokenAddress.toLowerCase(),
+                incentiveData.sIncentiveData.priceFeed,
+                incentiveData.sIncentiveData.priceFeedDecimals
+              ),
+            });
+          calculatedReserveIncentives.underlyingAsset = isBaseAsset
+            ? '0xeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeee'
+            : calculatedReserveIncentives.underlyingAsset; // ETH, MATIC, and AVAX all use reserve address of '0xee...'
+          allReserveIncentives.push(calculatedReserveIncentives);
+        }
+      });
+      setReserveIncentives(allReserveIncentives);
+    }
+  }, [reserves, reserveIncentiveData]);
+
+  if ((isRPCActive && rpcDataLoading) || (!isRPCActive && cachedDataLoading)) {
+    return <Preloader withBackground={true} />;
+  }
+
+  if (!activeData || (isRPCActive && rpcDataError) || (!isRPCActive && cachedDataError)) {
+    return <ErrorPage />;
+  }
+  // Compute the total claimable rewards for a user, returned as dictionary indexed by incentivesController
+  let userIncentives = calculateTotalUserIncentives({
+    reserveIncentives: reserveIncentiveData,
+    userReserveIncentives: userIncentiveData,
+    userReserves,
+    currentTimestamp,
+  });
+  userIncentives = Object.fromEntries(
+    Object.entries(userIncentives).filter(
+      (incentive) =>
+        incentive[1].rewardTokenAddress !== '0x0000000000000000000000000000000000000000'
```

Comment:
is this because of tribe? if yes please leave a comment in code as no-one will remember.

Reply:
This was to avoid displaying blank incentives. This was removed and replaced with this logic in IncentiveWrapper:

```
  // Only display assets for which user has claimable rewards
  const userIncentivesFiltered = Object.fromEntries(
    Object.entries(userIncentives).filter((entry) => Number(entry[1].claimableRewards) > 0)
  );
```